src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.models.FruitTransfer;
+import core.basesyntax.models.Operation;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    @Override
+    public void process(List<FruitTransfer> transferList,
+                        Map<Operation, OperationHandler> handlers) {
+        for (FruitTransfer lotOfFruit : transferList) {
+            OperationHandler chosenStrategy = handlers.get(lotOfFruit.getOperations());
+            if (chosenStrategy != null) {
+                chosenStrategy.performOperation(lotOfFruit);
+            }
```

Comment:
you should create a separate OperationStrategy class that returns Handler by Operation or throw an exception if the handler doesn’t  exist

Reply:
Why I should do so? Does it break some principles SOLID or other? Please explain it to me

Reply:
Let's discuss this in the chat Mate, I'm sure Lilia will explain her point of view. However, we want to throw an exception if "FruitTransfer" has an invalid operation, this is called the Fail Fast principle